lib/zip/entry.rb

```diff
@@ -577,10 +578,13 @@ def gather_fileinfo_from_srcpath(src_path) # :nodoc:
 
     def write_to_zip_output_stream(zip_output_stream) #:nodoc:all
       if @ftype == :directory
-        zip_output_stream.put_next_entry(self, nil, nil, ::Zip::Entry::STORED)
+        @compression_method = ::Zip::Entry::STORED
```

Comment:
Yes, I think I like that idea. I'll add that change.

Reply:
Another thing: with the change I've already made, which would be reinforced by this change as well, the `put_next_entry` form which uses separate parameters instead of just an `Entry` is not used at all within the library itself. I do wonder if we could get rid of it altogether if we're making major version changes anyway at this point?

I'm going to think a bit more on that idea.